kube-runtime/src/watcher.rs

```diff
@@ -640,19 +640,58 @@ pub fn watch_object<K: Resource + Clone + DeserializeOwned + Debug + Send + 'sta
 
 /// Default watch [`Backoff`] inspired by Kubernetes' client-go.
 ///
-/// Note that the exact parameters used herein should not be considered stable.
-/// The parameters currently optimize for being kind to struggling apiservers.
-/// See [client-go's reflector source](https://github.com/kubernetes/client-go/blob/980663e185ab6fc79163b1c2565034f6d58368db/tools/cache/reflector.go#L177-L181)
-/// for more details.
+/// This fn has been moved into [`DefaultBackoff`].
 #[must_use]
+#[deprecated(
+    since = "0.84.0",
+    note = "replaced by `watcher::DefaultBackoff`. This fn will be removed in 0.88.0."
+)]
 pub fn default_backoff() -> impl Backoff + Send + Sync {
-    let expo = backoff::ExponentialBackoff {
+    let bo = default_exponential_backoff();
+    ResetTimerBackoff::new(bo, DEFAULT_RESET_DURATION)
+}
+
+/// Default watcher backoff inspired by Kubernetes' client-go.
+///
+/// The parameters currently optimize for being kind to struggling apiservers.
+/// The exact parameters are taken from
+/// [client-go's reflector source](https://github.com/kubernetes/client-go/blob/980663e185ab6fc79163b1c2565034f6d58368db/tools/cache/reflector.go#L177-L181)
+/// and should not be considered stable.
+///
+/// This struct implements [`Backoff`] and is the default strategy used
+/// when calling `WatchStreamExt::default_backoff`. If you need to create
+/// this manually then [`DefaultBackoff::default`] can be used.
+pub struct DefaultBackoff(Strategy);
```

Comment:
RE name of the type, another idea (not sure if that's *actually* better):

```rust
pub mod backoff {
    pub fn default() -> Default {
        // actual constructor
    }

    pub struct Default(...); // no constructor, no Default impl
}
```

Usage: `watcher::backoff::default()`

Reply:
there is something nice about actually implementing `Default` though, you do get benefits such as 
https://github.com/kube-rs/kube/pull/1232/files#diff-d1bc7ffb915e37cefd8621bff0c7c5d9e19965babf0e4b7b02d8da4a8010a871R544

i.e. being able to say `Box::<DefaultBackoff>::default()` over `Box::new(DefaultBackoff::default())` because of how wrapper structs give you nice automatic implementation of default.